### PR TITLE
add bismark/summary

### PR DESCRIPTION
**Pull request by phue**
~Depends on https://github.com/nf-core/modules/pull/239 to generate the necessary test data~

## PR checklist

- [x] This comment contains a description of changes (with reason).
- [x] If you've fixed a bug or added code that should be tested, add tests!
- [x] If you've added a new tool - have you followed the module conventions in the [contribution docs](https://github.com/nf-core/modules/tree/master/.github/CONTRIBUTING.md)
- [x] If necessary, include test data in your PR.
- [x] Remove all TODO statements.
- [x] Emit the `<SOFTWARE>.version.txt` file.
- [x] Follow the naming conventions.
- [x] Follow the parameters requirements.
- [x] Follow the input/output options guidelines.
- [x] Add a resource `label`
- [x] Use BioConda and BioContainers if possible to fulfil software requirements.
- Ensure that the test works with either Docker / Singularity. Conda CI tests can be quite flaky:
    - [ ] `PROFILE=docker pytest --tag <MODULE> --symlink --keep-workflow-wd`
    - [ ] `PROFILE=singularity pytest --tag <MODULE> --symlink --keep-workflow-wd`
    - [ ] `PROFILE=conda pytest --tag <MODULE> --symlink --keep-workflow-wd`

### CODE REVIEW

File: software/bismark/summary/main.nf [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+
+    conda (params.enable_conda ? "bioconda::bismark=0.23.0" : null)
+    if (workflow.containerEngine == 'singularity' && !params.singularity_pull_docker_container) {
+        container "https://depot.galaxyproject.org/singularity/bismark:0.23.0--0"
+    } else {
+        container "quay.io/biocontainers/bismark:0.23.0--0"
+    }
+
+    input:
+    path(bam)
```
Comment (KevinMenden) [inline software/bismark/summary/main.nf:21]: sure this doesn't need a `meta` map?

**Review by KevinMenden — COMMENTED** [review]
LGTM, only thing is whether you want to add a `meta` map to that `bam` input?

File: software/bismark/summary/main.nf (hunk 1 above)
Comment (phue) [reply software/bismark/summary/main.nf:21]: `bismark_summary` produces a summary report over many samples, it just requires the bam files to infer sample names. We don't need the `meta` map in that case

**Review by grst — APPROVED** [review]
LGTM

**Pull request merged**